GameServerLib/GameObjects/Particle.cs

```diff
@@ -6,19 +6,26 @@ namespace LeagueSandbox.GameServer.GameObjects
 {
     public class Particle : GameObject, IParticle
     {
-        public IChampion Owner { get; }
+        public IGameObject Owner { get; }
         public string Name { get; }
         public string BoneName { get; }
         public float Size { get; }
 
-        public Particle(Game game, IChampion owner, ITarget t, string particleName, float size = 1.0f, string boneName = "", uint netId = 0)
+        public Particle(Game game, IGameObject owner, ITarget t, string particleName, float size = 1.0f, string boneName = "", uint netId = 0)
                : base(game, t.X, t.Y, 0, 0, netId)
         {
             Owner = owner;
             Target = t;
             Name = particleName;
             BoneName = boneName;
             Size = size;
+            _game.PacketNotifier.NotifyFXCreateGroup(this);
+        }
+
+        public void Remove()
+        {
+            SetToRemove();
+            _game.PacketNotifier.NotifyFXKill(this);
```

Comment:
I think `SetToRemove` call a function like `OnRemove`. If yes, please move `_game.PacketNotifier.NotifyFXKill(this);` to there (as we don't want to have sync problems).